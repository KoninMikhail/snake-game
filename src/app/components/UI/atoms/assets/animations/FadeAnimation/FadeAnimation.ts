import styled, { keyframes } from 'styled-components';
import { IAnimationPropsWithDirection } from '@ui/atoms/assets/animations/Animation.interface';

const FadeAnimationKeyframes = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

/**
 * @name FadeAnimation
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use for creation fade-in or fade-out effect
 *
 * @param delayBeforeChange - delayBeforeChange before animation
 * @param direction - animation direction
 * @return {JSX.Element}
 */
export const FadeAnimation = styled.div<IAnimationPropsWithDirection>`
    width: inherit;
    height: inherit;
    animation-duration: ${({ theme }) => theme.animations.fade.duration.toString() + 'ms'};
    animation-fill-mode: ${({ theme }) => theme.animations.fade.fill};
    animation-name: ${FadeAnimationKeyframes};
    animation-direction: ${({ direction }) => (direction === 'in' ? 'normal' : 'reverse')};
    animation-delay: ${({ delay = 0 }) => delay.toFixed().toString() + 'ms'};
    animation-iteration-count: ${({ theme }) => theme.animations.fade.count};
    animation-timing-function: ${({ theme }) => theme.animations.fade.timing};
`;
