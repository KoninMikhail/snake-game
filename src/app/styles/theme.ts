import {colors, ThemeColors} from './theme_parts/colors';
import {gaps, ThemeGaps} from './theme_parts/gaps';
import {icons, ThemeIcons} from './theme_parts/icons';
import {ThemeTypography, typography} from './theme_parts/typography';
import {animations, ThemeAnimations} from '@styles/theme_parts/animations';

/**
 * @typedef IBaseTheme
 */
export interface IBaseTheme {
    animations: ThemeAnimations;
    colors: ThemeColors;
    gaps: ThemeGaps;
    icons: ThemeIcons;
    typography: ThemeTypography;
}

/**
 * Theme
 */
export const theme: IBaseTheme = {
    colors,
    animations,
    gaps,
    icons,
    typography,
};
