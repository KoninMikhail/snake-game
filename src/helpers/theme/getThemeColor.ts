import { ThemeColors } from '@styles/theme_parts/colors';
import { IBaseTheme } from '@styles/theme';

/**
 * @name getThemeColor
 *
 * Get the color from theme.
 *
 * @param theme - The theme object
 * @param color - The color you want to get from the theme.
 * @returns A function that takes a theme and a color and returns the color from the theme.
 */
const getThemeColor: GetFromThemeFn<IBaseTheme, ThemeColors> = (theme, color) => {
    return theme.colors[color];
};

export default getThemeColor;
