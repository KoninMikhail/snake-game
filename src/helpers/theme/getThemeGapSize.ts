import {IBaseTheme} from '@styles/theme';
import {ThemeGaps} from '@styles/theme_parts/gaps';

/**
 * @name getThemeGapSize
 *
 * Get the gap size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the gap you want to get.
 * @returns A function that takes a theme and a size and returns the theme.gaps[size]
 */
const getThemeGapSize: GetFromThemeFn<IBaseTheme, ThemeGaps> = (theme, size) => theme.gaps[size];

export default getThemeGapSize;
