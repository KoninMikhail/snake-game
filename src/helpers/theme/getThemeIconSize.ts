import { IBaseTheme } from '@styles/theme';
import { ThemeIcons } from '@styles/theme_parts/icons';

/**
 * @name getThemeIconSize
 *
 * Get the icon size from the theme.
 *
 * @param theme - The theme object
 * @param size - The size of the icon you want to get.
 * @returns A function that takes a theme and a size and returns the theme.icons[size]
 */
const getThemeIconSize: GetFromThemeFn<IBaseTheme, ThemeIcons> = (theme, size) => {
    return theme.icons[size];
};

export default getThemeIconSize;
