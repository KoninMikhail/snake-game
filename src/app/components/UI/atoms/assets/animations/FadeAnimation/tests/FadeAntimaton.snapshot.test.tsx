import { render } from '@testing-library/react';
import { FadeAnimation } from '../FadeAnimation';
import { IBaseTheme } from '@styles/theme';

/**
 * Mocks
 */
const fakeTheme = {
    animations: {
        fade: {
            duration: 800,
            count: 1,
            timing: 'fade',
            fill: 'forwards',
        },
    },
} as unknown as IBaseTheme;

/**
 * Tests
 */
describe('FadeAnimation', () => {
    it('equal snapshot', () => {
        const { container } = render(
            <FadeAnimation direction={'in'} delay={500} theme={fakeTheme} />
        );
        expect(container).toMatchSnapshot();
    });
});
