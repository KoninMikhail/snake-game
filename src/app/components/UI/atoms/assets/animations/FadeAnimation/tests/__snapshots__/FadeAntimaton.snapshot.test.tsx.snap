// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeAnimation equal snapshot 1`] = `
.c0 {
  width: inherit;
  height: inherit;
  -webkit-animation-duration: 800ms;
  animation-duration: 800ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-name: cJoqxJ;
  animation-name: cJoqxJ;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-delay: 500ms;
  animation-delay: 500ms;
  -webkit-animation-iteration-count: 1;
  animation-iteration-count: 1;
  -webkit-animation-timing-function: fade;
  animation-timing-function: fade;
}

<div>
  <div
    class="c0"
    direction="in"
  />
</div>
`;
